src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -209,7 +209,7 @@ private SyntaxToken QuickScanSyntaxToken()
                 var flags = uc < charPropLength ? (CharFlags)s_charProperties[uc] : CharFlags.Complex;
 
                 state = (QuickScanState)s_stateTransitions[(int)state, (int)flags];
-                if (state == QuickScanState.Done || state == QuickScanState.Bad)
+                if (state >= QuickScanState.Done)
```

Comment:
This change is fragile because it takes dependency on the order of declaration within the enum. We should add an assert that ```QuickScanState.Bad == QuickScanState.Done + 1``` and add a comment in the enum that QuickScanState.Bad should be the last and should follow QuickScanState.Done and point where do we take advantage of that.